src/main/java/uk/gov/hmcts/darts/cache/token/service/AbstractTokenCache.java

```diff
@@ -107,16 +119,76 @@ public Optional<Token> store(CacheValue value, Boolean reuseTokenIfPossible) thr
         return token;
     }
 
-    private boolean reuseTokenBasedOnCredentials(Boolean reuseTokenIfPossible) {
-        return properties.isShareTokenForSameCredentials()
-                || (reuseTokenIfPossible != null && reuseTokenIfPossible);
+    @Override
+    @Transactional
+    @SuppressWarnings("java:S6809")
+    public Optional<Token> store(ServiceContext context) throws CacheException {
+        return store(context, false);
     }
 
-    protected abstract Predicate<String> getValidateToken();
-
     @Override
-    public Token getToken(String token) {
-        return Token.readToken(token, properties.isMapTokenToSession(), getValidateToken());
+    @Transactional
+    @SuppressWarnings("java:S6809")
+    public Optional<Token> store(ServiceContext context, Boolean reuseTokenIfPossible) throws CacheException {
+        Optional<Token> foundToken;
+
+        if (reuseTokenBasedOnCredentials(reuseTokenIfPossible)) {
+            log.info("Looking for the shared token configuration");
```

Comment:
Are these `log.info`s necessary to keep around for a bit - wondering if they should be debug or trace instead perhaps? Then does it need some parameters to be helpful

Reply:
yeah i ideally wanted something. I'll switch to debug for now